src/shared/components/Types/TypesIcon.tsx

```diff
@@ -13,20 +24,36 @@ const TypesIcon: React.SFC<TypesIconProps> = ({ type }) => {
   return (
     <ul className="types-list">
       {type.map((type, index) => {
+        const [colors, setColor] = React.useState<string[] | null>(null);
```

Comment:
I am not sure that you can use React hooks in a map like that, plus it's in the return statement of your component which I don't think is okay either.

You need to make sure you follow that https://reactjs.org/docs/hooks-rules.html

Reply:
Good to know! fixed it!